PA_implementation_overview.md

```diff
@@ -0,0 +1,66 @@
+# Protected Audience Implementation Overview
+This document is intended to give a rough overview of the Protected Audience implementation in Chrome, with a focus towards performance-relevant design decisions, and timeouts.  The purpose is to both explain how these work, and help consumers figure out if different timeouts/timeout behavior would be useful.
+
+This document starts out discussing singleton auctions, without any nested component auctions.  There is a section on component auctions at the end.
+
+
+## Phase 1 Load Interest Groups
+Immediately after runAdAuction() is invoked, all interest groups, k-anon data, and win history will be loaded from the SQLite database, wrapped by an in-memory cache to avoid duplicate lookups.  No worklets will be loaded until after this is complete, for all interestGroupBuyers participating in an auction.  Interest groups are loaded on a per-buyer basis, and remain grouped by buyer for the entire auction.  Interest groups that cannot participate in the auction (due to, e.g., no biddingLogicURL) are removed.  If there are no interest groups that can bid in the auction, and no additional bids, the auction completes without a winner, bypassing future stages.
```

Comment:
Removed all mentions of the term.  In most places, "script" is actually more correct than "script runner", since "worklet" is often used to refer to the script itself, as opposed to the script + its execution environment.